test/with_dummyserver/test_https.py

```diff
@@ -698,6 +698,27 @@ def test_tls_protocol_name_of_socket(self):
             finally:
                 conn.close()
 
+    def test_sslkeylogfile(self, tmpdir):
+        if not hasattr(ssl.SSLContext, "keylog_filename"):
+            pytest.skip("SSLContext does not support keylog_filename")
+        try:
+            keylog_file = tmpdir.join("keylogfile")
+            os.environ["SSLKEYLOGFILE"] = str(keylog_file)
```

Comment:
Can we use monkeypatch.setenv from pytest instead?

Reply:
ah, yes, monkeypatch is much nicer. Done.